background.ts

```diff
@@ -827,7 +827,10 @@ ipcMainProxy.on('k8s-restart', async() => {
 });
 
 ipcMainProxy.on('k8s-versions', async() => {
-  window.send('k8s-versions', await k8smanager.kubeBackend.availableVersions, await k8smanager.kubeBackend.cachedVersionsOnly());
+  const versions = await k8smanager.kubeBackend.availableVersions;
+  const cachedOnly = await k8smanager.kubeBackend.cachedVersionsOnly();
+
+  window.send('k8s-versions', versions.map(e => ({ version: e.version.version, channels: e.channels })), cachedOnly);
```

Comment:
Should `SemanticVersionEntry` have a `versionEntry` method that returns the data as a `VersionEntry` object, and then you could write

```suggestion
  window.send('k8s-versions', versions.map(e => e.versionEntry), cachedOnly);
```

Reply:
Currently `SemanticVersionEntry` is an interface, so it doesn't have any methods. I can look into switching it to a concrete class, though.